sys/net/application_layer/dns/dns.c

```diff
@@ -140,7 +140,7 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
                 ((_type == DNS_TYPE_AAAA) && (family == AF_INET)) ||
                 ! ((_type == DNS_TYPE_A) || ((_type == DNS_TYPE_AAAA))
                     )) {
-            if (((bufpos - pos) + addrlen) < (bufpos - pos)) {
+            if (addrlen < len) {
```

Comment:
The comparison needs to be the other way around.

Reply:
Need to go to bed -.-... Fixed.